config/opal_setup_cc.m4

```diff
@@ -179,33 +179,24 @@ AC_DEFUN([OPAL_SETUP_CC],[
     OPAL_C_COMPILER_VENDOR([opal_c_vendor])
 
     if test $opal_cv_c11_supported = no ; then
-        # It is not currently an error if C11 support is not available. Uncomment the
-        # following lines and update the warning when we require a C11 compiler.
-        # AC_MSG_WARNING([Open MPI requires a C11 (or newer) compiler])
-        # AC_MSG_ERROR([Aborting.])
-        # From Open MPI 1.7 on we require a C99 compliant compiler
-        dnl with autoconf 2.70 AC_PROG_CC makes AC_PROG_CC_C99 obsolete
-        m4_version_prereq([2.70],
-            [],
-            [AC_PROG_CC_C99])
-        # The result of AC_PROG_CC_C99 is stored in ac_cv_prog_cc_c99
-        if test "x$ac_cv_prog_cc_c99" = xno ; then
-            AC_MSG_WARN([Open MPI requires a C99 (or newer) compiler. C11 is recommended.])
-            AC_MSG_ERROR([Aborting.])
-        fi
-
-        # Get the correct result for C11 support flags now that the compiler flags have
-        # changed
-        OPAL_PROG_CC_C11_HELPER([], [], [])
+        # C11 is required
+        AC_MSG_WARN([Open MPI requires a C11 (or newer) compiler. C11 is required.])
+        AC_MSG_ERROR([Aborting.])
+    elif test $opal_cv_c11_flag_required = no ; then
+        # Check if the compiler accepts -std=c11
+        # this helps avoid accidentally using newer features
+        #opal_prog_cc_c11_flags="-std=c11 -c11"
+        #AC_MSG_NOTICE([checking if $CC supports C11 standard flag])
+        #for flag in $(echo $opal_prog_cc_c11_flags | tr ' ' '\n') ; do
+        #    _OPAL_CHECK_SPECIFIC_CFLAGS($flag, std_c11)
+        #    if test "x$opal_cv_cc_std_c11" == "x1" ; then
+        #        OPAL_FLAGS_APPEND_UNIQ([CFLAGS], ["$flag"])
+        #        AC_MSG_NOTICE([using $flag to ensure C11 support])
+        #        break
+        #    fi
+        #done
```

Comment:
If we're not going to use this code, go ahead and delete it.  It'll be in git history if we need to bring it back.